CHANGELOG.md

```diff
@@ -3,7 +3,191 @@
+
+The added `opa_eval` export comes with an ABI bump to version 1.2.
+See [#3627](https://github.com/open-policy-agent/opa/pull/3627) for all details.
+
+Along the same line, we've examined the processing of query evaluations that are Wasm-backed _through the `rego` package_.
+This allowed us to avoid unneccessary work ([#3666](https://github.com/open-policy-agent/opa/issues/3666)).
+
+
+#### Unsafe declared variables now cause a compile-time error
+
+Before this release, local variables that had been _declared_, i.e. introduced via the `some` keyword, had been able
+to slip through the safety checks unnoticed.
+
+For example, a policy like
+
+```rego
+package demo
+
+q { 
+	input == "open sesame"
+}
+
+p[x] {
+	some x
+}
+```
+
+would have _not_ caused any error **if `data.demo.p` wasn't queried**.
+Querying `data.demo.p` would return an "var requires evaluation" error.
+
+With this release, the erroneous rule no longer goes unnoticed, but is **caught at compile time**: "var x is unsafe".
+
+The most likely fix is to remove the rule with the unsafe variable, since it cannot have contributed to a successful
+evaluation in previous OPA versions.
+
+See [#3580](https://github.com/open-policy-agent/opa/issues/3580) for details.
+
+### Topdown and Rego
+
+- New built-in function: `crypto.x509.parse_and_verify_certificates` ([#3601](https://github.com/open-policy-agent/opa/issues/3601)), authored by @[jalseth](https://github.com/jalseth)
+
+  This function enables you to verify that there is a chain from a leaf certificate back to the trusted root.
+- New built-in function: `rand.intn` generates a random number between `0` and `n` ([#3615](https://github.com/open-policy-agent/opa/issues/3615)), authored by @[base698](https://github.com/base698)
+
+  The function takes a string argument to ensure that the same call, within one policy evaluation, returns the same random number.
+- `http.send` enhancement: New `caching_mode` parameter to configure if deserialized or serialized response bodies should be cached ([#3599](https://github.com/open-policy-agent/opa/issues/3599)) 
+- Custom built-in function enhancement: let custom builtins halt evaluation ([#3534](https://github.com/open-policy-agent/opa/issues/3534))
+- Partial evaluation: Fix stack overflow on certain expressions ([#3559](https://github.com/open-policy-agent/opa/issues/3559))
+
+### Tooling
+
+- Query Profiling: `opa eval --profile` now supports a `--count=#` flag to gather metrics and profiling data over multiple runs, and displays aggregate statistics for the results ([#3651](https://github.com/open-policy-agent/opa/issues/3651)).
+
+  This allows you to gather more robust numbers to assess policy performance.
+
+- Docker images: Publish static image ([#3633](https://github.com/open-policy-agent/opa/issues/3633))
+
+  As of this release, you can use the staticly-built Linux binary from a docker image: `openpolicyagent/opa:v0.31.0-static`.
```

Comment:
Remove the `v` from the image tag. This is just a legacy thing; the makefile doesn't include "v" in the image tags.